Mimir.Tests/QueryTests/CombinationSlotsTest.cs

```diff
@@ -0,0 +1,87 @@
+    {
+        var allCombinationSlotAddress = new Address("0x0000000001000000000200000000030000000004");
+        var allCombinationSlotState = new AllCombinationSlotState
+        {
+            CombinationSlots = new Dictionary<int, CombinationSlotState>
+            {
+                { 0, new CombinationSlotState { Index = 0 } },
+                { 1, new CombinationSlotState { Index = 1 } },
+            }
+        };
+        var mockRepo = new Mock<IAllCombinationSlotStateRepository>();
+        mockRepo
+            .Setup(repo => repo.GetByAddressAsync(It.IsAny<Address>()))
+            .ReturnsAsync(new AllCombinationSlotStateDocument(1, allCombinationSlotAddress, allCombinationSlotState));
+
+        var serviceProvider = TestServices.CreateServices(
+            allCombinationSlotStateRepositoryMock : mockRepo
+        );
+
+    var query = $$"""
+                      query {
+                        combinationSlots(avatarAddress: "{{allCombinationSlotAddress}}") {
+                          key
+                          value {
+                            address
+                            index
+                            isUnlocked
+                            petId
+                            startBlockIndex
+                            unlockBlockIndex
+                            result {
+                              tradableFungibleItemCount
+                              typeId
+                              costume {
+                                elementalType
+                                equipped
+                                grade
+                                id
+                                itemId
+                                itemSubType
+                                itemType
+                                requiredBlockIndex
+                                spineResourcePath
+                              }
+                              itemUsable {
+                                elementalType
+                                grade
+                                id
+                                itemId
+                                itemSubType
+                                itemType
+                                requiredBlockIndex
+                              }
+                              tradableFungibleItem {
+                                elementalType
+                                grade
+                                id
+                                itemId
+                                itemSubType
+                                itemType
+                                requiredBlockIndex
+                              }
+                            }
+                          }
+                        }
+                      }                 
+                    """;
+        var result = TestServices.ExecuteRequestAsync(
+            serviceProvider,
+            b => b.SetQuery(query)
+        );
+
+        await Verify(result);
+    }
+}
```

Comment:
Could you add empty line to end of file?

Reply:
Sure!